libs/@local/hash-graph-sdk/python/turbo.json

```diff
@@ -19,7 +19,7 @@
       "dependsOn": ["codegen:blocking", "codegen:filter"]
     },
     "build": {
-      "dependsOn": ["^build", "codegen"],
+      "dependsOn": ["^build", "codegen", "lint:mypy"],
```

Comment:
It's the other way around. It's not required to lint a package to build it but the linting may have a different output if the dependencies failed.

Reply:
mh yes, well, my thought about this was/is the following: `mypy` is the build step for Python as it ensures that the code will not be malformed (this is also why only `lint:mypy` is included, not `lint:black` or `lint:ruff`) as mypy is our best chance to verify that something _will_ run in production, even if the build step (which doesn't exist) successfully runs. Hope that makes sense?

Reply:
I get your point, but this is precisely what linting is about. There is no point in lint every project in your dependency chain each time, it is required only once. Other packages require a package to be build but they don't care how it ensures it's correctness.

Saying it from another perspective: you wouldn't want to require `clippy` to pass to run your app. Especially you don't want to require upstream linting to pass.

Reply:
> `mypy` is the build step for Python 

No, it's a linter. Building would be packing it into a wheel.

Reply:
That's not what I meant, and you know it... and I wouldn't put `clippy` on the same level as `mypy`, `clippy` is more in line with `ruff`. **anyway** this doesn't really matter and as requested I have inverted the relationship in [f471b3e](https://github.com/hashintel/hash/pull/3122/commits/f471b3ea6d8e8e30d9b7e438f91bebf398b8f82d).

`mypy` is a lot more in line with `tsc` from typescript and even though one can only transpile, `hash-ai-worker-ts` for example chooses to type-check on build as well ([see here](https://github.com/hashintel/hash/blob/main/apps/hash-ai-worker-ts/package.json))

Reply:
That's true, `clippy` is probably more comparable to `ruff`, yet, `mypy` (just like `tsc`) is _not_ required to build a package. We don't use the build-step for many things at the moment, but we have defined the dependencies via it. If I run `yarn lint:eslint` for example, I don't want `yarn lint:mypy` to be called in a few packages as well, because it's not needed for that.

Reply:
Yes, `mypy` is not required just like `tsc`, but often just a sanity check^^ in any case if in future the `build` step gets to be more important one should maybe think about unifying this. Either by removing `tsc` (and just transpiling) which would be a lot faster, or including `mypy`.